### ISSUE TITLE
[BUG] Folderless .gpg files cause crash in auto fill search

### ISSUE THREAD

**Issue by Drainful** [android-password-store/Android-Password-Store#700]
**Describe the bug**
On the latest snapshot version when using the auto-fill search feature I get an uncaught exception. I have folderless .gpg files titled after their services (eg. github.com.gpg) as imported from bitwarden with pass import. Example file:
```
password123
login: mail@example.com
url: https://github.com/login
```
I am perfectly willing to change the format of my files, but the ability to auto-fill from top-level .gpg files seems useful, and at the very least a crash should be avoided.

Stack trace: https://pastebin.com/8SwA5pw4

The issue does not occur if there are no top level .gpg files.

**To Reproduce**
1. add a file named github.com.gpg to the password store outside of any directory
2. visit github.com and attempt login
3. use the autofill search feature 
4. press the X button to clear the existing search
5. insert "g" or any other character to trigger the crash

**Expected behavior**
github.com.gpg is found by the search and is usable for autofill

**Device information (please complete the following information):**
 - Device: Nokia 3.1
 - OS: stock ROM Android version 9.0
 - App version: Snapshot 2020-04-13 17:28

**Additional context**
Ideally the url field in the .gpg file could be used for auto-fill so that the search is unnecessary

**Comment by FabianHenneke** [comment]
Thanks for the detailed report! This should be fixed with #703, which will be in the snapshot build once this issue has been closed.

> Ideally the url field in the .gpg file could be used for auto-fill so that the search is unnecessary

Unfortunately, by design, this is not really possible: Since the contents of the .gpg file are encrypted, this would require decrypting every (top-level) .gpg file in the repository. At best, this would be very slow, but if the GPG private key is stored on an external hardware token (e.g. a YubiKey), this would be downright impossible. The best I can offer you is a manual search the first time you use the credential, followed by using the stored Autofill match from there on. @msfjarvis Please speak up if I should have missed a better solution.

**Comment by msfjarvis** [comment]
No I think you've got the same solution I'd have come up with. Storing results in autofill is much more reliable and pragmatic than trying to decrypt every single password, especially when that involves both APS and OpenKeychain to work.
